slack-api-client/src/main/java/com/slack/api/methods/impl/FilesUploadV2Helper.java

```diff
@@ -106,12 +107,17 @@ public FilesUploadV2Response completeUploads(
 
         result.setFiles(new ArrayList<>());
         for (FilesCompleteUploadExternalResponse.FileDetails file : response.getFiles()) {
-            FilesInfoResponse fileInfo = this.client.filesInfo(r -> r.token(v2Request.getToken()).file(file.getId()));
-            underlyingException.getFileInfoResponses().add(fileInfo);
-            if (!fileInfo.isOk()) {
-                throw underlyingException;
+            if (v2Request.isRequestFileInfo()) {
```

Comment:
NIT: I would tend to try and place this in its own function named something like `fetchFile` that returns the `File` object from `fileInfo.getFile()` or `partialFileObject`

Reply:
Thanks for the suggestion. In this case, the part won't be reused elsewhere plus we have to pass v2Request 
 and underlyingException to the internal method, which is not so simple. So, let me go with the current code.